### DISCUSSION TITLE
[Bug]: fs data not exporting to influxdb

### DISCUSSION THREAD

**Discussion by Garulf** [nicolargo/glances#3223] [General]
Before filling this issue, please read the manual (https://glances.readthedocs.io/en/latest/) and search if the bug do not already exists in the database (https://github.com/nicolargo/glances/issues).

#### Description

When running C:\Python27\python.exe -m glances --export-influxdb
No data is ever exported to influxdb regarding fs.

The measurements and fields are created however just never populated with any data.

These are the keys created by glances in influxdb: https://pastebin.com/YACf5FXS

#### Versions

* Glances (glances -V): v2.10
* PSutil (glances -V): v5.2.2
* Operating System (lsb_release -a): Windows 10

#### Logs
Logs: https://pastebin.com/g0Q18NqM

**Comment by Garulf** [comment]
Glances _DOES_ export correctly to influxdb, however due to the field keys using special characters (":" and "\\") it is unable to retrieve the data in any of the API calls usable.

I've tried double quotes, single quotes and escaping special characters.

Google searches and all documentation,  have no relevant information.

Is there any way to change the names Glances uses?

**Comment by nicolargo** [comment]
Hi @Akrasia , thanks for the issue report.

Glances can not change the key name, the FS key is the drive letter (C:\ for Windows). Have you tried to add quote to your API request: "C:\" ? (have a look on the InfluxDB issue comment: https://github.com/influxdata/influxdb/issues/6011#issuecomment-196547163)

**Comment by Garulf** [comment]
Influxdb seems to have a lot of issues if slashes or other non-word characters are in the key ([a-zA-Z0-9_]).

The proper command should be:

`influx -execute 'select "C:\\.used" FROM fs'` <--returns nothing

However the character for "ALL" works.

`influx -execute 'select * FROM fs'` <---returns all keys, showing that data is saved... just not accessible individually

Solution?:
Refrain from using non-word characters when naming keys. I don't really see a need for a colon and back slash when the drive letter is all that's needed.

Thank you for responding.

**Comment by ACSmith1337** [comment]
This is still an issue.
